contracts/core/State.sol

```diff
@@ -93,6 +94,9 @@ contract State is Constants, Objects, ReentrancyGuard, Ownable {
     uint32 public twaiLength;
     /**** new interest model end */
 
+    mapping(address => VolumeTracker.Observation[65535]) internal volumeTradedObservations; //recorded Observations for every trade per user
```

Comment:
shall we put it in a struct ? will it be more efficient?

Reply:
all 3 mappings

Reply:
I think it has the same efficiency at best because there is no compression going on